### PR TITLE
Improvements for crash report

### PR DESCRIPTION
**Pull request by dgmachado**

### CODE REVIEW

File: NativeScript/runtime/Helpers.mm [hunk 1]
```diff
@@ -691,3 +691,13 @@
     Log(@"%s", stack.c_str());
     assert(false);
 }
+
+void tns::StopExecutionAndLogStackTraceIfInDebug(v8::Isolate* isolate) {
```
Comment (edusperoni) [inline NativeScript/runtime/Helpers.mm:695]: rename to `StopExecutionAndLogStackTrace` or remove

File: NativeScript/runtime/Helpers.mm [hunk 2]
```diff
@@ -691,3 +691,13 @@
...
+    Assert(false, isolate);
+}
+
+void tns::LogDebugMessage(std::string message) {
+    if (RuntimeConfig.IsDebug) {
+        Log(@"%s", message.c_str());
+    }
+}
```
Comment (edusperoni) [inline NativeScript/runtime/Helpers.mm:703]: remove and use `Log` directly

File: NativeScript/runtime/MetadataBuilder.mm [hunk 3]
```diff
@@ -810,6 +810,14 @@
     Class klass = objc_getClass(containingClass.c_str());
     // TODO: Find out if the isMethodCallback property can be determined based on a UITableViewController.prototype.viewDidLoad.call(this) or super.viewDidLoad() call
 
+    NSString* message = [NSString stringWithFormat:@"MetadataBuilder::InvokeMethod class {%s}, selector {%s}, isInitializer {%s}, type {%s}, lib {%s}",
+                         containingClass.c_str(),
+                         meta->selectorAsString(),
+                         meta->isInitializer() ? "true":"false",
+                         meta->typeName(),
+                         meta->topLevelModule()->getName()];
+    tns::LogDebugMessage(message.UTF8String);
```
Comment (edusperoni) [inline NativeScript/runtime/MetadataBuilder.mm:819]:
```
if(debug) {
   NSString* message = [NSString stringWithFormat:@"MetadataBuilder::InvokeMethod class {%s}, selector {%s}, isInitializer {%s}, type {%s}, lib {%s}",
                         containingClass.c_str(),
                         meta->selectorAsString(),
                         meta->isInitializer() ? "true":"false",
                         meta->typeName(),
                         meta->topLevelModule()->getName()];
    tns::Log(message.UTF8String);
}
```

if you don't check for debug first, you'll waste processing on building the message and never show it

File: NativeScript/runtime/Helpers.mm (hunk 2 above)
Comment (edusperoni) [reply NativeScript/runtime/Helpers.mm:703]: to clarify. This function will induce people to make the mistake of building a big heavy string and discarding it here.

File: NativeScript/runtime/Helpers.mm (hunk 1 above)
Comment (dgmachado) [reply NativeScript/runtime/Helpers.mm]: Done

File: NativeScript/runtime/MetadataBuilder.mm (hunk 3 above)
Comment (dgmachado) [reply NativeScript/runtime/MetadataBuilder.mm]: Done

**Pull request merged**